js/plugins/iq.js

```diff
@@ -192,145 +203,111 @@ var IqPlugin= function (framework)
+          this.ooyalaReporter.reportPlayerLoad();
+          OO.log("IQ: Reported: reportPlayerLoad()");
         }
         break;
       //OO.EVENTS.INITIAL_PLAY -> OO.Analytics.EVENTS.VIDEO_PLAY_REQUESTED.
       case OO.Analytics.EVENTS.INITIAL_PLAYBACK_REQUESTED:
-        /* TODO: disable for now as this is already reported by reporter.js in core */
-        //OO.log("IQ: Reported: reportPlayRequested() with args: " + autoPlay);
-        //this.ooyalaReporter.reportPlayRequested(autoPlay);
+        OO.log("IQ: Reported: reportPlayRequested() with args: " + autoPlay);
+        this.ooyalaReporter.reportPlayRequested(autoPlay);
         break;
       //OO.EVENTS.PLAYHEAD_TIME_CHANGED -> OO.Analytics.EVENTS.VIDEO_STREAM_POSITION_CHANGED.
       case OO.Analytics.EVENTS.VIDEO_STREAM_POSITION_CHANGED:
-        if (params && params[0])
-        {
+        if (params && params[0] && params[0].streamPosition > 0)
+        {            
           currentPlayheadPosition = params[0].streamPosition;
-          if (currentPlayheadPosition > 0)
+          if (playingInstreamAd)
           {
-            if (this.ooyalaReporter)
-            {
-              /* TODO: disable for now as this is already reported by reporter.js in core */
-              //var currentPlayheadPositionMilli = currentPlayheadPosition * 1000;
-              //this.ooyalaReporter.reportPlayHeadUpdate(currentPlayheadPositionMilli);
-              //OO.log("IQ: Reported: reportPlayHeadUpdate() with args: " + Math.floor(currentPlayheadPosition * 1000));
-            }
-            else
+            var totalTime = params[0].totalStreamDuration;
+            var percentPlayed = 0;
+            var reportQuartile = false;
+            if (totalTime != null && totalTime > 0 )
             {
-              OO.log("IQ: Tried reporting event: " + OO.Analytics.EVENTS.VIDEO_STREAM_POSITION_CHANGED +
-                     " but ooyalaReporter is: " + this.ooyalaReporter);
+              if (currentPlayheadPosition >= 0.25 * totalTime && !adFirstQuartile){
+                percentPlayed = 0.25;
+                adFirstQuartile = true;
+                reportQuartile = true;
+              } 
+              else if (currentPlayheadPosition >= 0.50 * totalTime && !adSecondQuartile)
+              {
+                percentPlayed = 0.50;
+                adSecondQuartile = true;
+                reportQuartile = true;
+              } 
+              else if (currentPlayheadPosition >= 0.75 * totalTime && !adThirdQuartile)
+              {
+                percentPlayed = 0.75;
+                adThirdQuartile = true;
+                reportQuartile = true;
+              } 
+              else if (currentPlayheadPosition >= 1.0 * totalTime && !adLastQuartile)
+              {
+                percentPlayed = 1.00;
+                adLastQuartile = true;
+                reportQuartile = true;
```

Comment:
if we start a video 3/4 of the way in (using initialTime) should we also send the 1st and 2nd quartile messages?  You may want to chat with Juan but I think we always assume we will have the other quartiles as well.

Reply:
@pilievOoyala is the scrubber bar available during ads? Note that these quartiles are only for ad playthrough reporting... for the actual video we will continually call reportPlayHeadUpdate events for each update of the playhead.

Reply:
@pilievOoyala is correct, if we report the third quartile but haven't reported the first and second, we'll need to send those as well.

Reply:
@aeng7 @pilievOoyala the current code will send the 1st quartile for the first VIDEO_STREAM_POSITION_CHANGED received during the ad if it is started 3/4 of the way in. The if-else-if chain here relies on the adXQuartile bools to keep track of which quartiles have already been sent. So even if the ad is moved after the 3rd quartile, the 1st quartile will still be sent at this time since currentPlayheadPosition >= 0.25 * totalTime  and !adFirstQuartile are both true (since adFirstQuartile has not been set to true yet). The next update of the playhead will send the 2nd quartile, next one will do the 3rd, etc.

One problem I do see: it might be that if the ad starts too late there won't be enough playhead updates to "catch up" in quartile reporting. I could change it so that this event sends all quartiles that the playhead has passed immediately, if that is the expected behavior.

Reply:
@dallen42 , let's make this change. This could help avoid a few potential headaches down the road (however unlikely these headaches are)